File: solutions/cpp/task_003.cpp
#include <vector>
#include <iostream>
#include <cassert>
#include <algorithm>

int FindOutlier(std::vector<int> arr)
{
  std::vector<int> even;
  std::vector<int> odd;
  int even_cnt = 0;
  int odd_cnt = 0;
  for(auto it = arr.begin(); it != arr.end(); it++){
    if(abs(*it)%2==1){
      odd_cnt++;
      odd.push_back(*it);
    }else{
      even_cnt++;
      even.push_back(*it);
    }
  }
  if(even_cnt > odd_cnt)
    return odd[0];
  return even[0];
}

int main(){
    assert(FindOutlier({2, 3, 4}) == 3);
    assert(FindOutlier({1, 2, 3}) == 2);
    assert(FindOutlier({4, 1, 3, 5, 9}) == 4);

    std::cout << "All tests pass." << std::endl;

    std::vector<int> t{2, 3, 4};
    std::cout << "Outliner for numbers: ";
    std::for_each(t.begin(), t.end(), [](const auto &e){std::cout<< e << " ";});
    std::cout << "is: " << FindOutlier(t) << std::endl;

    t = std::vector<int>{1, 2, 3};
    std::cout << "Outliner for numbers: ";
    std::for_each(t.begin(), t.end(), [](const auto &e){std::cout<< e << " ";});
    std::cout << "is: " << FindOutlier(t) << std::endl;

    t = std::vector<int>{4, 1, 3, 5, 9};
    std::cout << "Outliner for numbers: ";
    std::for_each(t.begin(), t.end(), [](const auto &e){std::cout<< e << " ";});
    std::cout << "is: " << FindOutlier(t) << std::endl;
}

File: solutions/python/task_003.py
def FindOutlier(arr):
    even = []
    odd = []
    even_cnt = 0
    odd_cnt = 0
    for item in arr:
        if item % 2 == 1:
            odd_cnt+=1
            odd.append(item)
        else:
            even_cnt+=1
            even.append(item)

    if even_cnt > odd_cnt:
        return odd[0]
    return even[0]



assert FindOutlier([1, 2, 3]) == 2
assert FindOutlier([4, 1, 3, 5, 9]) == 4
assert FindOutlier([2, 3, 4]) == 3

print('All tests pass.')

arr = [1, 2, 3]
print('Outliner for numbers: [{}] is {}'.format(' '.join([str(x) for x in arr]), FindOutlier(arr)))

arr = [4, 1, 3, 5, 9]
print('Outliner for numbers: [{}] is {}'.format(' '.join([str(x) for x in arr]), FindOutlier(arr)))

arr = [2, 3, 4]
print('Outliner for numbers: [{}] is {}'.format(' '.join([str(x) for x in arr]), FindOutlier(arr)))